pkg/exporter/azureblob_exporter.go

```diff
@@ -19,23 +19,60 @@ const (
 )
 
 // AzureBlobExporter defines an Azure Blob Exporter
-type AzureBlobExporter struct{}
+type AzureBlobExporter struct {
+	BaseExporter
+}
 
 var _ interfaces.Exporter = &AzureBlobExporter{}
 
-// Export implements the interface method
-func (exporter *AzureBlobExporter) Export(files []string) error {
-	APIServerFQDN, err := utils.GetAPIServerFQDN()
-	if err != nil {
-		return err
+// NewAzureBlobExporter is a constructor
+func NewAzureBlobExporter() *AzureBlobExporter {
+	return &AzureBlobExporter{
+		BaseExporter: BaseExporter{
+			exporterType: AzureBlob,
+		},
 	}
+}
+
+// GetStorageContainerName get storage container name
+func (exporter *AzureBlobExporter) GetStorageContainerName(APIServerFQDN string) (string, error) {
+	var containerName string
+	var err error
+	if utils.IsKubernetesInDocker() {
+		containerName, err = exporter.GetKubernetesInDockerStorageContainerName(APIServerFQDN)
+	} else {
+		containerName, err = exporter.GetNonKINDStorageContainerName(APIServerFQDN)
+	}
+
+	//TODO run a sanitizer over the final chars in the containerName
+	return containerName, err
+}
+
+func (exporter *AzureBlobExporter) GetKubernetesInDockerStorageContainerName(APIServerFQDN string) (string, error) {
+	return APIServerFQDN, nil
```

Comment:
I tested periscope on the kubeadm cluster, which will also call this function along with the kind cluster. Setting the container like this will give errors such as "the resource contains invalid characters". Could you modify this function to the following: 
```
func (exporter *AzureBlobExporter) GetKubernetesInDockerStorageContainerName(APIServerFQDN string) (string, error) {

        containerName := strings.Replace(APIServerFQDN, ".", "-", -1)

	return containerName, nil
}
```

Reply:
will do - again if you have one handy it would be great if you could provide an example of a kubeadm apiServerFQDN and the containerName you would expect as result, so I can use as test data

Reply:
also - can you confirm that on a kubeadm cluster we will call this function due to the kubeconfig being located in "/etc/kubernetes/kubelet.conf"? And hence the helper function currently named "IsKubernetesInDocker" will return true?

If the answer to the above is yes, it should probably necessitate a rename or minor redesign to this function

Reply:
This is the example for the kubeadm cluster I'm using to test: 
{apiServerFQDN: "troubleshoot.testing.db839748.eastus.cloudapp.azure.com", containername: "troubleshoot-testing-db839748-eastus-cloudapp-azure-com"}

Reply:
And yes, the kubeadm cluster has the kubeconfig located in "/etc/kubernetes/kubelet.conf", so the IsKubernetesInDocker boolean will return true.